### PR TITLE
[DOCS] Update Checkpoint and Actions Reference with testing

### PR DESCRIPTION
**Pull request by NathanFarmer**
Changes proposed in this pull request:
- GREAT-379

### Definition of Done
- [X] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [X] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [X] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

**Review by alexsherstinsky — APPROVED** [review]
LGTM (as long as one additional reviewer re-reads the content with a fine-tooth comb).  Thanks, @NathanFarmer !

File: tests/integration/docusaurus/reference/core_concepts/checkpoints_and_actions.py [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+  - name: store_validation_result
+    action:
+      class_name: StoreValidationResultAction
+  - name: store_evaluation_params
+    action:
+      class_name: StoreEvaluationParametersAction
+  - name: update_data_docs
+    action:
+      class_name: UpdateDataDocsAction
+evaluation_parameters:
```
Comment (donaldheppner) [inline tests/integration/docusaurus/reference/core_concepts/checkpoints_and_actions.py:270]: Is there a way to assert that the evaluation parameters as configured worked as expected?

File: tests/integration/docusaurus/reference/core_concepts/checkpoints_and_actions.py [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+config_version: 1
+class_name: Checkpoint
+run_name_template: "%Y-%M-foo-bar-template-$VAR"
+validations:
+  - batch_request:
+      datasource_name: taxi_datasource
+      data_connector_name: default_inferred_data_connector_name
+      data_asset_name: yellow_tripdata_sample_2019-01
+      data_connector_query:
+        index: -1
```
Comment (donaldheppner) [inline tests/integration/docusaurus/reference/core_concepts/checkpoints_and_actions.py]: This index: -1 just looks broken to me when I read it. Is it required for the example to work? If we could remove this thing I'd consider it a win; open to some discussion on this point

File: docs/reference/checkpoints_and_actions.md [hunk 3]
```diff
@@ -136,303 +136,132 @@ example, the `run_results` dictionary from a WarningAndFailureExpectationSuiteCh
...
+
+Below is an example of a `CheckpointResult` object which itself contains `ValidationResult`, `ExpectationSuiteValidationResult`, and `CheckpointConfig` objects.
+
+#### Example CheckpointResult:
+
+```python file=file=../../../../tests/integration/docusaurus/reference/core_concepts/checkpoints_and_actions.py#L102-L116
 ```
 
-## Checkpoint configuration examples
+## Checkpoint configuration default and override behavior
```
Comment (donaldheppner) [inline docs/reference/checkpoints_and_actions.md:149]: Can we include a line like: "The code in the following examples is under test and can be found at: [github link to file]"?

**Review by donaldheppner — CHANGES_REQUESTED** [review]

File: docs/reference/checkpoints_and_actions.md (hunk 3 above)
Comment (NathanFarmer) [reply docs/reference/checkpoints_and_actions.md:149]: You are correct that this is missing, but the standard way we have used to communicate this is to add the link at the bottom in "Additional Notes". The link has been added.

File: tests/integration/docusaurus/reference/core_concepts/checkpoints_and_actions.py (hunk 2 above)
Comment (NathanFarmer) [reply tests/integration/docusaurus/reference/core_concepts/checkpoints_and_actions.py]: It's not required. I went back and forth on this as well. It is telling the data connector to use the most recent batch in the data asset, but that is the default behavior regardless so I will remove it.

Comment (NathanFarmer) [reply tests/integration/docusaurus/reference/core_concepts/checkpoints_and_actions.py]: I have removed this option from all configs in this doc.

File: tests/integration/docusaurus/reference/core_concepts/checkpoints_and_actions.py (hunk 1 above)
Comment (NathanFarmer) [reply tests/integration/docusaurus/reference/core_concepts/checkpoints_and_actions.py:270]: I have added assertions for evaluation parameters

File: docs/reference/checkpoints_and_actions.md (hunk 3 above)
Comment (donaldheppner) [reply docs/reference/checkpoints_and_actions.md:149]: Great, I think linking to this great work publicly will really help people make best use of the docs

**Review by donaldheppner — APPROVED** [review]

**Pull request merged**